icx-http-server/src/main.rs

```diff
@@ -116,19 +120,58 @@ async fn forward_request(
 
     let entire_body = body::to_bytes(request.into_body()).await?.to_vec();
 
-    HttpRequestCanister::create(agent.as_ref(), canister_id)
+    let canister = HttpRequestCanister::create(agent.as_ref(), canister_id.clone());
+    let (http_response,) = canister
         .http_request(method, uri.to_string(), headers, &entire_body)
         .call()
         .await
-        .map_err(Into::<Box<dyn Error>>::into)
-        .and_then(|(http_response,)| {
-            let mut builder =
-                Response::builder().status(StatusCode::from_u16(http_response.status_code)?);
-            for HeaderField(name, value) in http_response.headers {
-                builder = builder.header(&name, value);
+        .map_err(Into::<Box<dyn Error>>::into)?;
+
+    let mut builder = Response::builder().status(StatusCode::from_u16(http_response.status_code)?);
+    for HeaderField(name, value) in http_response.headers {
+        builder = builder.header(&name, value);
+    }
+
+    if let Some(mut token) = http_response.next_token {
+        let (mut sender, body) = body::Body::channel();
+        let agent = agent.as_ref().clone();
+        sender.send_data(Bytes::from(http_response.body)).await?;
+
+        tokio::spawn(async move {
+            let canister = HttpRequestCanister::create(&agent, canister_id);
+            // We already call it once.
+            let mut count = 1;
+            loop {
+                count += 1;
+                if count >= MAX_HTTP_REQUEST_NEXT_CALL_COUNT {
```

Comment:
If `MAX_HTTP_REQUEST_NEXT_CALL_COUNT` means "max number of calls to `http_request_next`, then:
```suggestion
            // We have not yet called http_request_next.
            let mut count = 0;
            loop {
                count += 1;
                if count > MAX_HTTP_REQUEST_NEXT_CALL_COUNT {
```

Example: if `MAX_HTTP_REQUEST_NEXT_CALL_COUNT` were 0, this would allow 0 calls to `http_request_next`, if 1, then 1, if 2, then 2.

As written, if `MAX_HTTP_REQUEST_NEXT_CALL_COUNT` were 2, this would allow 0 calls to `http_request_next`, and 1 chunk of data total, because `count` would be 2 on the first `>=` check.

Reply:
Good call.. That will show me to code late...